pkg/sentry/fsimpl/fuse/fusefs.go

```diff
@@ -258,12 +260,20 @@ func (fs *filesystem) newInode(nodeID uint64, attr linux.FUSEAttr) *kernfs.Dentr
 	i := &inode{fs: fs, NodeID: nodeID}
 	creds := auth.Credentials{EffectiveKGID: auth.KGID(attr.UID), EffectiveKUID: auth.KUID(attr.UID)}
 	i.InodeAttrs.Init(&creds, linux.UNNAMED_MAJOR, fs.devMinor, fs.NextIno(), linux.FileMode(attr.Mode))
+	// TODO(gvisor.dev/issue/1193): This should be handled in Init,
+	// unfortunately we need it for this implementation now.
+	atomic.StoreUint64(&i.size, attr.Size)
 	i.OrderedChildren.Init(kernfs.OrderedChildrenOptions{})
 	i.dentry.Init(i)
 
 	return &i.dentry
 }
 
+// Ino overrides the Ino() from InodeAttrs.
+func (i *inode) Ino() uint64 {
+	return i.NodeID
+}
+
```

Comment:
I don't recommend doing this since `Ino()` is the original function provided by `kernfs` and it returns the inode number returned by the FUSE server, which is supposed to be different from `NodeID` of FUSE (a identifier of FUSE object). Though we only care about `NodeID` when issuing FUSE request, but the inode number resides in the attributes is still exposed in `stat(2)`.

For example, inode number of 8654226 is fixed no matter what NodeID is generated to refer to this object. They are different and both needed.
```
root@dc3b3d59ce69:/fus# stat nomount/bin
  File: nomount/bin -> usr/bin
  Size: 7               Blocks: 0          IO Block: 4096   symbolic link
Device: 40h/64d Inode: 8654226     Links: 1
Access: (0777/lrwxrwxrwx)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2020-07-20 14:43:46.000000000 +0000
Modify: 2020-07-20 14:43:46.000000000 +0000
Change: 2020-08-05 01:05:09.528636719 +0000
 Birth: -
```

Reply:
In this case, please check if my modifications to Stat() broke this or not.